File: src/content/docs/topics/cs/coding-problems/stack/853-car-fleet.go
package main

import "fmt"

func assert(condition bool, msgs ...string) {
	if !condition {
		msg := "assertion failed"
		if len(msgs) > 0 {
			msg = msgs[0]
		}
		panic(msg)
	}
}

func carFleet(target int, position []int, speed []int) int {
	// TODO: implement
	return 0
}

func runTests() {
	assert(carFleet(12, []int{10, 8, 0, 5, 3}, []int{2, 4, 1, 1, 3}) == 3)
	assert(carFleet(10, []int{3}, []int{3}) == 1)
	assert(carFleet(100, []int{0, 2, 4}, []int{4, 2, 1}) == 1)
	assert(carFleet(10, []int{6, 8}, []int{3, 2}) == 2)
	fmt.Println("all tests pass")
}

func main() { runTests() }

File: src/content/docs/topics/cs/coding-problems/stack/853-car-fleet.ts
function assert(condition: boolean, msg: string = ''): void {
    if (!condition) throw new Error(msg || 'Assertion failed');
}

function carFleet(target: number, position: number[], speed: number[]): number {
    // TODO: implement
    return 0;
}

function _runTests(): void {
    assert(carFleet(12, [10, 8, 0, 5, 3], [2, 4, 1, 1, 3]) === 3);
    assert(carFleet(10, [3], [3]) === 1);
    assert(carFleet(100, [0, 2, 4], [4, 2, 1]) === 1);
    assert(carFleet(10, [6, 8], [3, 2]) === 2);
    // perf
    const t0 = performance.now();
    carFleet(500_001, Array.from({ length: 500_000 }, (_, i) => i), new Array(500_000).fill(1));
    console.log(`perf 500000-car fleet: ${(performance.now() - t0).toFixed(1)}ms`);
    console.log('all tests pass');
}

_runTests();
